integration-tests/tests/src/tests/mixed.ts

```diff
@@ -220,23 +345,882 @@ describe("Mixed", () => {
+              onEnd();
+            }
+          };
+        };
+
+        /**
+         * Helper function to use as a generic handler for dictionary listeners.
+         * @param onEnd Callback to run once the listener reaches the end of the changes array. If the callback is Mocha.Done, it will ensure that
+         * the listener was run exactly [changesOnRun.length] times
+         * @param changesOnRun An array of Realm.DictionaryChangeSet.
+         * The handler will assert equality of listener's changes with the next element of the array on every run.
+         */
+        const expectDictionaryChangesOnEveryRun = (
+          onEnd: () => void | Mocha.Done,
+          changesOnRun: Realm.DictionaryChangeSet[],
+        ) => {
+          return (_: Realm.Dictionary<any>, changes: Realm.DictionaryChangeSet) => {
+            runCount++;
+
+            const expectedChanges = changesOnRun[runCount - 1];
+            expect(changes.insertions).members(expectedChanges.insertions);
+            expect(changes.modifications).members(expectedChanges.modifications);
+            expect(changes.deletions).members(expectedChanges.deletions);
+
+            if (runCount >= changesOnRun.length) {
+              // Once runCount reaches given array, run the onEnd callback. If onEnd is Mocha.done then
+              // running it multiple times will cause tests to fail so this also ensures the listener fires
+              // exactly changesOnRun.length times.
+              onEnd();
+            }
+          };
+        };
+
+        /**
+         * Helper function to use as a generic handler for object listeners.
+         * @param onEnd Callback to run once the listener reaches the end of the changes array. If the callback is Mocha.Done, it will ensure that
+         * the listener was run exactly [changesOnRun.length] times
+         * @param changesOnRun An array of Realm.ObjectChangeSet.
+         * The handler will assert equality of listener's changes with the next element of the array on every run.
+         */
+        const expectObjectChangesOnEveryRun = <T extends Realm.Object<T>>(
+          onEnd: () => void | Mocha.Done,
+          changesOnRun: Realm.ObjectChangeSet<T>[],
+        ) => {
+          return (_: Realm.Object<T>, changes: Realm.ObjectChangeSet<T>) => {
+            runCount++;
+
+            const expectedChanges = changesOnRun[runCount - 1];
+            expect(changes.deleted).equals(expectedChanges.deleted);
+            expect(changes.changedProperties).members(expectedChanges.changedProperties);
+
+            if (runCount >= changesOnRun.length) {
+              // Once runCount reaches given array, run the onEnd callback. If onEnd is Mocha.done then
+              // running it multiple times will cause tests to fail so this also ensures the listener fires
+              // exactly changesOnRun.length times.
+              onEnd();
+            }
+          };
+        };
```

Comment:
I believe this could be simplified by using the [src/utils/sequence.ts](https://github.com/realm/realm-js/blob/main/integration-tests/tests/src/utils/sequence.ts) or perhaps even the [src/utils/listener-stub.ts](https://github.com/realm/realm-js/blob/main/integration-tests/tests/src/utils/listener-stub.ts).

Reply:
This PR or another, either would be fine by me, as long as it was on your radar 👍